### PR TITLE
SPCTR-3354 Resolved PHPError for `preg_match` and `strlen` due to global  `$_wp_current_template_content` getting null in non FSE theme with PHP 8.1.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://wordpress.org/support/topic/preg_match-passing-null-and-strlen-passing-null/

### Screenshots

- https://d.pr/v/e5qMU6
- https://d.pr/v/IpjUQS

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the task.
- Also mention test case from this task - https://brainstormforce.atlassian.net/browse/SPCTR-2722

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-post-assets.php [hunk 1]
```diff
@@ -385,13 +385,15 @@ public function allow_assets_generation() {
...
-		// If the current template has content and contains blocks, execute this code block.
-		if ( has_blocks( $this->post_id ) || ( $_wp_current_template_content && has_blocks( $blocks ) ) ) {
-			/* Print conditional css for all blocks */
-			add_action( 'wp_head', array( $this, 'print_conditional_css' ), 80 );
+		if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+			global $_wp_current_template_content;
+			$blocks = parse_blocks( $_wp_current_template_content );
+			// Global Required assets.
+			// If the current template has content and contains blocks, execute this code block.
+			if ( has_blocks( $this->post_id ) || ( $_wp_current_template_content && has_blocks( $blocks ) ) ) {
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:393]: can you please verify if $this->post_id is available the also do we need to check it for block theme, also should this add_action be out of block theme check.

Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: @adi3890 Right, Resolved.

File: classes/class-uagb-post-assets.php [hunk 2]
```diff
@@ -385,11 +385,16 @@ public function allow_assets_generation() {
...
+			$blocks                              = array();
+			$spectra_wp_current_template_content = '';
+		if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+			global $_wp_current_template_content;
+			$spectra_wp_current_template_content = $_wp_current_template_content;
+			$blocks                              = parse_blocks( $spectra_wp_current_template_content );
+		}
+			// Global Required assets.
+			// If the current template has content and contains blocks, execute this code block.
+		if ( has_blocks( $this->post_id ) || ( $spectra_wp_current_template_content && has_blocks( $blocks ) ) ) {
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:397]:
```suggestion
		if ( has_blocks( $this->post_id ) || has_blocks( $blocks ) ) {
```
@AbhijitNage123 do we need to check for $spectra_wp_current_template_content as we are getting blocks from it only.
- alos i am curious if there can be a case thta current theme is block theme but $_wp_current_template_content is still null.

Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php:397]: @adi3890 In block theme, $_wp_current_template_content never return null, it always has an value in block theme.

Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: Resolved.

File: classes/class-uagb-post-assets.php [hunk 3]
```diff
@@ -385,11 +385,16 @@ public function allow_assets_generation() {
...
-		global $_wp_current_template_content;
-		$blocks = parse_blocks( $_wp_current_template_content );
-		// Global Required assets.
-		// If the current template has content and contains blocks, execute this code block.
-		if ( has_blocks( $this->post_id ) || ( $_wp_current_template_content && has_blocks( $blocks ) ) ) {
+			$blocks                              = array();
+			$spectra_wp_current_template_content = '';
+		if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+			global $_wp_current_template_content;
+			$spectra_wp_current_template_content = $_wp_current_template_content;
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:392]: as we have remove this value from check why do we need this variable?
Please review this code once remove unneccessary code.

File: classes/class-uagb-post-assets.php [hunk 4]
```diff
@@ -385,11 +385,16 @@ public function allow_assets_generation() {
 	 * @since 1.23.0
 	 */
 	public function enqueue_scripts() {
-		global $_wp_current_template_content;
-		$blocks = parse_blocks( $_wp_current_template_content );
-		// Global Required assets.
-		// If the current template has content and contains blocks, execute this code block.
-		if ( has_blocks( $this->post_id ) || ( $_wp_current_template_content && has_blocks( $blocks ) ) ) {
+			$blocks                              = array();
+			$spectra_wp_current_template_content = '';
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:389]: we may not need this now.

File: classes/class-uagb-post-assets.php (hunk 3 above)
Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: Resolved

File: classes/class-uagb-post-assets.php (hunk 4 above)
Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: Resolved

**Review by adi3890 — APPROVED** [review]

**Review by rajmohanreddy774 — APPROVED** [review]

**Pull request merged**